Sources/xcresultparser/CoberturaCoverageConverter.swift

```diff
@@ -31,6 +31,68 @@
 import Foundation
 import XCResultKit
 
+enum JSONParseError: Error {
+    case convertError(code: Int, message: String)
+}
+
+// Subrange information struct
+struct Subrange: Codable {
+    let column: Int
+    let executionCount: Int
+    let length: Int
+}
+
+// LineDetail information struct
+struct LineDetail: Codable {
+    let isExecutable: Bool
+    let line: Int
+    let executionCount: Int?
+    let subranges: [Subrange]?
+}
+
+// FileCoverage information struct
+struct FileCoverage: Codable {
+    let files: [String: [LineDetail]]
+
+    // Custom initializer to handle the top-level dictionary
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        var filesDict = [String: [LineDetail]]()
+
+        for key in container.allKeys {
+            let keyString = key.stringValue
+            let lineDetails = try container.decode([LineDetail].self, forKey: key)
+            filesDict[keyString] = lineDetails
+        }
+        self.files = filesDict
+    }
+
+    // Custom encoding method to handle the top-level dictionary
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        for (key, value) in files {
+            guard let codingKey = CodingKeys(stringValue: key) else {
+                continue
+            }
+            try container.encode(value, forKey: codingKey)
+        }
+    }
```

Comment:
If you switch to only using `Decodable`, you can then remove this custom encoding method.
```suggestion
```